src/rpcblockchain.cpp

```diff
@@ -1063,1370 +1097,2389 @@ int64_t AmountFromDouble(double dAmount)
+    // Send
+    CReserveKey keyChange(pwalletMain);
+    int64_t nFeeRequired = 0;
+    bool fCreated = pwalletMain->CreateTransaction(vecSend, wtx, keyChange, nFeeRequired);
+    printf("Transaction Created.");
+
+    if (!fCreated)
+    {
+        if (totalAmount + nFeeRequired > pwalletMain->GetBalance())
+            throw JSONRPCError(RPC_WALLET_INSUFFICIENT_FUNDS, "Insufficient funds");
+        throw JSONRPCError(RPC_WALLET_ERROR, "Transaction creation failed");
     }
-    else if (sItem == "advertisebeacon")
+
+    printf("Committing.");
+    // Rain the recipients
+    if (!pwalletMain->CommitTransaction(wtx, keyChange))
     {
-                std::string sOutPubKey = "";
-                std::string sOutPrivKey = "";
-                std::string sError = "";
-                std::string sMessage = "";
+        printf("Commit failed.");
 
-                bool fResult = AdvertiseBeacon(sOutPrivKey,sOutPubKey,sError,sMessage);
-                entry.push_back(Pair("Result",SuccessFail(fResult)));
-                entry.push_back(Pair("CPID",GlobalCPUMiningCPID.cpid.c_str()));
-                entry.push_back(Pair("Message",sMessage.c_str()));
+        throw JSONRPCError(RPC_WALLET_ERROR, "Transaction commit failed");
+    }
+    std::string sNarr = "Rain successful:  Sent " + wtx.GetHash().GetHex() + ".";
+    printf("Success %s",sNarr.c_str());
 
-                if (!sError.empty())        entry.push_back(Pair("Errors",sError));
+    res.push_back(Pair("Response", sNarr));
 
-                if (!fResult)
-                {
-                    entry.push_back(Pair("FAILURE","Note: if your wallet is locked this command will fail; to solve that unlock the wallet: 'walletpassphrase <yourpassword> <240>'."));
-                }
-                if (fResult)
-                {
-                    entry.push_back(Pair("Public Key",sOutPubKey.c_str()));
-                    entry.push_back(Pair("Warning!","Your public and private research keys have been stored in gridcoinresearch.conf.  Do not lose your private key (It is non-recoverable).  It is recommended that you back up your gridcoinresearch.conf file on a regular basis."));
-                }
-                results.push_back(entry);
+    return res;
+}
 
-    }
-    else if (sItem == "syncdpor2")
-    {
-        std::string sOut = "";
-        bool bFull = GetCountOf("beacon") < 50 ? true : false;
-        LoadAdminMessages(bFull,sOut);
-        FullSyncWithDPORNodes();
-        entry.push_back(Pair("Syncing",1));
-        results.push_back(entry);
-    }
-    else if(sItem=="gatherneuralhashes")
-    {
-        GatherNeuralHashes();
-        entry.push_back(Pair("Sent","."));
-        results.push_back(entry);
+Value unspentreport(const json_spirit::Array& params, bool fHelp)
+{
+    if (fHelp || params.size() != 0)
+        throw runtime_error(
+                "unspentreport\n"
+                "\n"
+                "Displays unspentreport\n");
```

Comment:
This is not very explanatory

Reply:
yea, havent touched all them yet to add detail. didnt want to get distracted from the goal but will be updating